src/sys/windows.rs

```diff
@@ -469,6 +469,38 @@ pub(crate) fn recv_from(
     }
 }
 
+pub(crate) fn peek_sender(socket: Socket) -> io::Result<SockAddr> {
+    // Safety: `recvfrom` initialises the `SockAddr` for us.
+    let ((), sender) = unsafe {
+        SockAddr::try_init(|storage, addrlen| {
+            let res = syscall!(
+                recvfrom(
+                    socket,
+                    //
```

Comment:
```suggestion
```

Reply:
Heh, I wanted to say something there because it looks wrong passing a null pointer, but I wasn't sure what. Windows apparently doesn't care.